src/System.Management.Automation/engine/parser/Parser.cs

```diff
@@ -1620,15 +1620,30 @@ private ScriptBlockAst NamedBlockListRule(Token lCurly, List<UsingStatementAst>
                                             ? lCurly.Extent
                                             : (paramBlockAst != null) ? paramBlockAst.Extent : null;
             IScriptExtent endExtent = null;
-            IScriptExtent extent;
+            IScriptExtent extent = null;
+            IScriptExtent scriptBlockExtent = null;
 
             while (true)
             {
                 Token blockNameToken = NextToken();
                 switch (blockNameToken.Kind)
                 {
                     default:
+                        // Next token is unexpected.
+                        // ErrorRecovery: if 'lCurly' is present, pretend we saw a closing curly; otherwise, eat the unexpected token.
+                        if (lCurly != null) { UngetToken(blockNameToken); }
+
+                        // If 'lCurly == null", then it's a ps1/psm1 file, and thus the extent is the whole file.
+                        scriptBlockExtent = lCurly != null ? ExtentOf(startExtent, endExtent) : _tokenizer.GetScriptExtent();
```

Comment:
We already check `lCurly != null` in line 1634. Maybe use one "if"?

Reply:
Fixed.